### PR TITLE
Xor Of Sum Of All Pairs.cpp

### PR DESCRIPTION
**Pull request by souravchandra1910**
Added Xor Of Sum Of All Pairs in cpp

#### Checklist

- [x] Added description of change
- [ ] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [x] Relevant documentation/comments is changed or added
- [ ] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes:

### CODE REVIEW

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
Hey there! Thank you for submitting a PR! 😄 Sadly, this does not meet our requirements. Check out our [repository standards](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md). Please read them **carefully** and **follow them**. **Let us know if you need any help here or on our [Discord server](https://the-algorithms.com/discord).** 🙂

**Comment by souravchandra1910** [comment]
I have made the changes

**Comment by souravchandra1910** [comment]
Please see through

File: bit_manipulation/Xor Of Sum Of All Pairs.cpp [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+*And if we XOR all the elements first, then double the result, 
+*we've introduced a zero in the binary to the right of the answer.
+*Let's try to code this :
+*/
+
+// Time Complexity : O(N)
+// Space Complexity : O(1)  (no extra space) 
+
+#include <bits/stdc++.h>
+using namespace std;
```
Comment (Panquesito7) [inline bit_manipulation/Xor Of Sum Of All Pairs.cpp:44]: These are discouraged. Please use each header/library you need and strictly use `std::`.

File: bit_manipulation/Xor Of Sum Of All Pairs.cpp [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+class Solution
+{
+public:
+  int xor_sum(vector<int> &v1)
```
Comment (Panquesito7) [inline bit_manipulation/Xor Of Sum Of All Pairs.cpp:48]: Functions should be documented explaining the use of its parameters, a brief description, and what it returns.

File: bit_manipulation/Xor Of Sum Of All Pairs.cpp [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+  vector<int> v3 = {3, 5, 6, 2};
+  if (s.xor_sum(v3) == 4)
+    cnt++;
+  vector<int> v4 = {12, 54, 2, 2};
+  if (s.xor_sum(v4) == 116)
+    cnt++;
+  vector<int> v5 = {1, 5, 2, 1, 2};
+  if (s.xor_sum(v5) == 10)
+    cnt++;
+  if (cnt == 5)
```
Comment (Panquesito7) [inline bit_manipulation/Xor Of Sum Of All Pairs.cpp:81]: Test cases should be made using `assert` (requires adding the `cassert` library/header).

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
1. Please rename the file accordingly to match snake_case.
2. Explain the use of the algorithm in sections and the correct use of `*`. For example:

```c++
/**
 * @file
 * @brief Implementation of the [<algorithm>](<Wikipedia link or source for algorithm explanation>)
 *
 * @details
 * <Detailed description of the algorithm>
 * <Space complexity and other technical stuff>
 *
 * @author [<name>](<GitHub link)
 */
```

3. Put the functions into namespaces (except for `main` and the tests).
4. Other details can be found at our [**contributing guidelines**](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md).

If you need any help, let us know here or on our [**Discord server**](https://the-algorithms.com/discord)! 🙂

**Pull request closed without merging**